File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int i = 0;
        Set<Integer> set = new HashSet<>();
        while(i < nums.length){
            set.add(nums[i]);
            i++;
        }
        i = 0;
        int result = 0;
        while(i < nums.length){
            int temp = nums[i];
            int curr = 0;
            if(!set.contains(temp - 1)){
                while(set.contains(temp)){
                    curr++;
                    result = Math.max(result, curr);
                    temp++;
                }
            }
            i++;
        }
        return result;

    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result = 0
        hashset = set()
        for num in nums:
            hashset.add(num)
        for num in hashset:
            curr = 0
            if num - 1 not in hashset:
                while num in hashset:
                    curr = curr + 1
                    num = num + 1
                result = max(result, curr)
        return result

   


    


  